commands/verify/verify.go

```diff
@@ -87,6 +88,38 @@ func verifySingleCase(expectedFile, actualFile, query string) error {
 	return nil
 }
 
+type ErrContain struct {
+	errAddr *multierror.Error
+	mutex   sync.Mutex
+}
+
+func concurrentVerifySingleCase(idx int, v config.VerifyCase, errContain *ErrContain, retry int, interval time.Duration, wg *sync.WaitGroup) {
+	if v.GetExpected() == "" {
+		errMsg := fmt.Sprintf("the expected data file for case[%v] is not specified\n", idx)
+		logger.Log.Warnf(errMsg)
+		errContain.mutex.Lock()
+		errContain.errAddr = multierror.Append(errContain.errAddr, errors.New(errMsg))
+		errContain.mutex.Unlock()
+		wg.Done()
+		return
+	}
+
+	for current := 1; current <= retry; current++ {
+		if err := verifySingleCase(v.GetExpected(), v.GetActual(), v.Query); err == nil {
+			break
+		} else if current != retry {
+			logger.Log.Warnf("verify case[%v] failure, will continue retry, %v", idx, err)
+			time.Sleep(interval)
+		} else {
+			errContain.mutex.Lock()
+			errContain.errAddr = multierror.Append(errContain.errAddr, err)
+			errContain.mutex.Unlock()
+		}
+	}
```

Comment:
Besides, there is a tricky thing you need to consider: the fail-fast (or not) in concurrent mode.